src/main/java/com/google/devtools/build/lib/rules/proto/ProtoCommon.java

```diff
@@ -179,21 +180,18 @@ private ProtoCommon() {
   // TODO(lberki): Would be nice if had these in ProtoInfo instead of that haphazard set of fields
   // Unfortunately, ProtoInfo has a Starlark interface so that requires a migration.
   static final class Library {
-    private final ImmutableList<Artifact> sources;
     private final ImmutableList<Pair<Artifact, String>> importPathSourcePair;
     private final String sourceRoot;
 
     Library(
-        ImmutableList<Artifact> sources,
         String sourceRoot,
         ImmutableList<Pair<Artifact, String>> importPathSourcePair) {
-      this.sources = sources;
       this.sourceRoot = sourceRoot;
       this.importPathSourcePair = importPathSourcePair;
     }
 
     public ImmutableList<Artifact> getSources() {
-      return sources;
+      return ImmutableList.copyOf(Iterables.transform(importPathSourcePair, p -> p.first));
```

Comment:
nit, opt: you can replace `p -> p.first` with `Pair::getFirst` (I'm not sure if it's an improvement)